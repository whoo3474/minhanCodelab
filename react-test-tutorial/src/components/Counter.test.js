import React, { Component } from 'react';
import renderer from 'react-test-renderer';
import Counter from './Counter';

describe('Counter', () => {
  let component = null;

  it('renders correctly', () => {
    component = renderer.create(<Counter />);
  });

  it('matches snapshot', () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })

  //increase가 잘되는지 확인
  it('increases correctly', () => {
    component.getInstance().onIncrease();
    expect(component.getInstance().state.value).toBe(2); // value값이 2인지 확인
    const tree = component.toJSON(); // re-render
    expect(tree).toMatchSnapshot();
  });

  //decrease가 잘되는지 확인
  it('decreases correctly', () => {
    component.getInstance().onDecrease();
    expect(component.getInstance().state.value).toBe(1);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })
});