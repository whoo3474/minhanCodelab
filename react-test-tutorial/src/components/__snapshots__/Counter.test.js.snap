// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter decreases correctly 1`] = `
<div>
  <h1>
    카운터!!
  </h1>
  <h3>
    1
  </h3>
  <button
    onClick={[Function]}
  >
    +
  </button>
  <button
    onClick={[Function]}
  >
    -
  </button>
</div>
`;

exports[`Counter increases correctly 1`] = `
<div>
  <h1>
    카운터!!
  </h1>
  <h3>
    2
  </h3>
  <button
    onClick={[Function]}
  >
    +
  </button>
  <button
    onClick={[Function]}
  >
    -
  </button>
</div>
`;

exports[`Counter matches snapshot 1`] = `
<div>
  <h1>
    카운터!!
  </h1>
  <h3>
    1
  </h3>
  <button
    onClick={[Function]}
  >
    +
  </button>
  <button
    onClick={[Function]}
  >
    -
  </button>
</div>
`;
